### DISCUSSION TITLE
I would like to open a particular browser automatically for specific URLs or from specific apps

### DISCUSSION THREAD

**Discussion by will-stone** [will-stone/browserosaurus#476] [Q&A] [answered]

**Comment by will-stone** [comment, marked as answer]
Browserosaurus does not provide a method for automatically selecting browsers, rather opting for the consistency that the picker window will show every time. If you would like to be able to set rules, I highly recommend [Finicky](https://github.com/johnste/finicky) by [John Sterling](https://johnste.github.io/) which can very easily be used in tandem with Browserosaurus, meaning the best of both worlds: a rules engine _and_ a graphical interface.

**Reply by dcunited08** [reply]
Any recommended setup?  Finicky -> Browserosaurus or Browserosaurus -> Finicky?  At first thought, it seems Finicky can be set up as the primary with the rules and forward to Browserosaurus for everything else.

**Reply by mhowardISP** [reply]
I haven't tried this but I might because Choosy looks to be abandoned. I think you would do Finicky as your default browser and when it doesn't match any rules you'd make Browserosaurus the browser that is chosen

**Reply by notDavid** [reply]
Fyi for others, set Finicky as your default browser, then add something like this to your `.finicky.js` config ([also see here](https://github.com/johnste/finicky#example-configuration)) to open Browserosaurus for specific urls:

```
  {
    match: finicky.matchHostnames([ /.*signal.org$/,
                                    /.*proton.me$/,
									/.*mozilla.com$/,
                                  ]),
    browser: "Browserosaurus"
  },
```

**Reply by mvalezy** [reply]
Thank you :)

**Reply by raccmonteiro** [reply]
you can set Browserosaurus has default browser and everything that does not match any pattern will open Browserosaurus.
You can use this [https://finicky-kickstart.vercel.ap](https://finicky-kickstart.vercel.app) to config Finicky

**Comment by bricin** [comment]
Seems like the Finicky code as a pre-check for the Browsersaurus graphical launcher is a 4x coffee buyer to me ;-) Wish I had any Swift knowledge lurking around.

**Comment by christianjuth** [comment]
@will-stone Finicky says it's no longer actively developed. I understand your reason for wanting specific behavior, but it would be super cool if you could enable some sort of advanced mode that allows you to have apps like slack automatically open when you click a slack link.

I would even pay for this advanced mode if you put it behind a one time paywall.

**Reply by will-stone** [reply]
@christianjuth I mean this with peace and love, but it's not cool to offer money for features. How much would you be willing to pay to commission this work? [There's a great article on open source etiquette about this](https://teejeetech.medium.com/etiquette-in-open-source-projects-f302b6e19c38).

All that aside, Finicky is a great app, and I use it daily. Just because John isn't currently developing it, doesn't mean it doesn't work well.
